scheduler/reconcile_util.go

```diff
@@ -583,24 +584,36 @@ type allocNameIndex struct {
 	job, taskGroup string
 	count          int
 	b              structs.Bitmap
+
+	// duplicates is used to store duplicate allocation indexes which are
+	// currently present within the index tracking. The map key is the index,
+	// and the current count of duplicates. While the map is only accessed
+	// within a single routine, lock is used for access to provide future-proof
+	// safety at a very small overhead.
```

Comment:
Changing the scheduler goroutine to be multi-threaded would be an enormous change because so much of placement work is serialized on a per-job basis. And having a mutex that we have to lock and unlock potentially many times per scheduler invocation seems fairly expensive relative to the vague future value. I would leave this off.

Reply:
Resolved within commit https://github.com/hashicorp/nomad/pull/18873/commits/9fb64df1a75835ef9201cf1668f2890254c9ebcb